### ISSUE TITLE
Automatic Document Feeder Scan fails

### ISSUE THREAD

**Issue by MasterG87** [sbs20/scanservjs#296]
**Describe the bug**
Error from automatic feed scans even though the scanimage command works successfull if used manually.

**To Reproduce**
1. Scanservjs used with automatic document feeder produces an error, but manual scanimage command works.
4. See error from /var/sys/log for scanservjs via Chrome Browser:
`
Jun  2 08:51:00 tomraspbian server.js[532]: [2021-06-02T06:51:00.481Z] ERROR (Http): Error: /usr/bin/scanimage -d 'pixma:04A91759_32BD26' --mode 'Color' --source 'Automatic Document Feeder' --resolution 75 -l 0 -t 0 -x 210 -y 297 --format 'tiff' --batch=./data/temp/~tmp-scan-1-%04d.tif exited with code: 9, stderr: Scanning infinity pages, incrementing by 1, numbering from 1
Jun  2 08:51:00 tomraspbian server.js[532]: Scanning page 1
Jun  2 08:51:00 tomraspbian server.js[532]: Scanned page 1. (scanner status = 5)
Jun  2 08:51:00 tomraspbian server.js[532]: Scanning page 2
Jun  2 08:51:00 tomraspbian server.js[532]: Scanned page 2. (scanner status = 5)
Jun  2 08:51:00 tomraspbian server.js[532]: Scanning page 3
Jun  2 08:51:00 tomraspbian server.js[532]: scanimage: sane_read: Error during device I/O
Jun  2 08:51:00 tomraspbian server.js[532]: Scanned page 3. (scanner status = 9)
Jun  2 08:51:00 tomraspbian server.js[532]: Batch terminated, 3 pages scanned
Jun  2 08:51:00 tomraspbian server.js[532]:     at ChildProcess.proc.on (/var/www/scanservjs/server/process.js:54:18)
Jun  2 08:51:00 tomraspbian server.js[532]:     at ChildProcess.emit (events.js:198:13)
Jun  2 08:51:00 tomraspbian server.js[532]:     at maybeClose (internal/child_process.js:982:16)
Jun  2 08:51:00 tomraspbian server.js[532]:     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
`
See console output from manual command:
`
/usr/bin/scanimage -d 'pixma:04A91759_32BD26' --mode 'Color' --source 'Automatic Document Feeder' --resolution 75 -l 0 -t 0 -x 210 -y 297 --format 'tiff' --batch=test3.tiff
Scanning infinity pages, incrementing by 1, numbering from 1
Scanning page 1
Scanned page 1. (scanner status = 5)
Scanning page 2
Scanned page 2. (scanner status = 5)
Scanning page 3
scanimage: sane_read: Document feeder out of documents
Scanned page 3. (scanner status = 7)
Batch terminated, 3 pages scanned
`

**Expected behavior**
Expected that automatic document feeder scans produce a successful scan.

**Client (please complete the following information):**
 - Browser Chrome on Macbook
 - Version 90.0.4430.212

**Server (please complete the following information):**
 - OS: Raspbian 10 (buster) on a Rasberry pi
 - Node version 10.24.0
 - NPM version 7.11.2
 - Docker version not installed

**Logs**
[errorlog_scanservjs.txt](https://github.com/sbs20/scanservjs/files/6582747/errorlog_scanservjs.txt)

**Comment by sbs20** [comment]
What happens if you run `/usr/bin/scanimage -d 'pixma:04A91759_32BD26' --mode 'Color' --source 'Automatic Document Feeder' --resolution 75 -l 0 -t 0 -x 210 -y 297 --format 'tiff' --batch=scan-1-%04d.tif`?

If that works, can you try becoming the `scanservjs` user and trying again?

**Comment by MasterG87** [comment]
When I run the proposed manual command everything seems to work:
`/usr/bin/scanimage -d 'pixma:04A91759_32BD26' --mode 'Color' --source 'Automatic Document Feeder' --resolution 75 -l 0 -t 0 -x 210 -y 297 --format 'tiff' --batch=scan-1-%04d.tif

Scanning infinity pages, incrementing by 1, numbering from 1
Scanning page 1
Scanned page 1. (scanner status = 5)
Scanning page 2
Scanned page 2. (scanner status = 5)
Scanning page 3
scanimage: sane_read: Document feeder out of documents
Scanned page 3. (scanner status = 7)
Batch terminated, 3 pages scanned`

When I run the same command as user scanservjs the following error occurs:
`scanservjs@tomraspbian:~ $ /usr/bin/scanimage -d 'pixma:04A91759_32BD26' --mode 'Color' --source 'Automatic Document Feeder' --resolution 75 -l 0 -t 0 -x 210 -y 297 --format 'tiff' --batch=scan-1-%04d.tif

Scanning infinity pages, incrementing by 1, numbering from 1
Scanning page 1
Scanned page 1. (scanner status = 5)
Scanning page 2
Scanned page 2. (scanner status = 5)
Scanning page 3
scanimage: sane_read: Error during device I/O
Scanned page 3. (scanner status = 9)
Batch terminated, 3 pages scanned`

Do you have an idea?

**Comment by MasterG87** [comment]
Ok I think I know the reason. Scanservjs uses the old version 1.0.27 and my default user uses the new version 1.0.32 (manually compiled by myself). 
`scanservjs@tomraspbian:~ $ scanimage --version
scanimage (sane-backends) 1.0.32; backend version 1.0.27
scanservjs@tomraspbian:~ $ exit
exit
pi@tomraspbian:~ $ scanimage --version
scanimage (sane-backends) 1.0.32; backend version 1.0.32
`
So it is not a bug of your scanservjs. Sorry.

**Comment by sbs20** [comment]
Oh well done for finding that. Please don't be sorry. This is really useful information and will help others. With any luck this will fix itself once Bullseye comes out.

**Comment by MasterG87** [comment]
I have found another issue with the options automatic document feeder and multiple pages scan on, unluckily.
I am using the latest version 2.15.1 of your tool. Before using a config file (config.local.js) everything worked fine. Then I wanted to comment in one line in the config.default.js and name it as config.local.js. After that change the only line I commented in is config.ocrLanguage = 'deu';

When I now scan only one page (multiple page option turned off), your tool works fine an the language is detected correctly. But when I use the multiple pages option on, I get an error from the scanimage command (see error log attached). Do you have an idea? I also tried to remove the config.local.js file, but the error remains. I am confused now.

Manual command works using user scanservjs:
`/usr/bin/scanimage -d 'pixma:04A91759_32BD26' --mode 'Color' --source 'Automatic Document Feeder' --resolution 300 -l 0 -t 0 -x 210 -y 297 --format 'tiff' --batch=tmp-scan-1-%04d.tif

Scanning infinity pages, incrementing by 1, numbering from 1
Scanning page 1
Scanned page 1. (scanner status = 5)
Scanning page 2
Scanned page 2. (scanner status = 5)
Scanning page 3
scanimage: sane_read: Document feeder out of documents
Scanned page 3. (scanner status = 7)
Batch terminated, 3 pages scanned`

[scanservjs.log.installed.txt](https://github.com/sbs20/scanservjs/files/6601005/scanservjs.log.installed.txt)

**Comment by sbs20** [comment]
It looks like the error you're getting is:

```
Jun 04 21:58:14 tomraspbian server.js[529]: scanimage: sane_read: Error during device I/O
Jun 04 21:58:14 tomraspbian server.js[529]: Scanned page 3. (scanner status = 9)
Jun 04 21:58:14 tomraspbian server.js[529]: Batch terminated, 3 pages scanned
```

which is exactly the same error you were getting previously. I'm not sure how you reconfigured the scanservjs user to use the later version of SANE, was it an environment variable / path? Is it possible that that has reset back to the standard version for buster?

The `config.local.js` change wouldn't have caused a problem - at least not like this - as it's an error at a much lower level than the software can control. But in order to make the changes come into effect, you would have had to restart the service which in turn could have caused it to pick up different environment variables.

Is that a possibility?

**Comment by MasterG87** [comment]
Ok thanks for your reply. I don't know why but somehow scanservjs wasn't using my LD_LIBRARY_PATH necessary for my newer version of sane-backends. Previously, I did put it into my .bashrc and it worked. No I had to remove it from there and put it into your config file. Now everything works fine, without using the .bashrc. Thanks for pointing me into the right direction.

The bold line solved my problem with adding the LD_LIBRARY_PATH directly to the config file.
`   config.ocrLanguage = 'deu';
     config.log.level = 'DEBUG';
     **config.scanimage = 'LD_LIBRARY_PATH=/usr/local/lib /usr/local/bin/scanimage';**
     config.convert = '/usr/bin/convert';
     config.tesseract = '/usr/bin/tesseract';
     `
